packages/core/typescript/TypeScriptFileUpdate.ts

```diff
@@ -65,64 +66,120 @@ export class TypeScriptFileUpdate {
-						// found routes variable
-						node = ts.visitEachChild(node, visitRoutesVariable, context);
-					} else {
-						node = ts.visitEachChild(node, visitor, context);
-					}
-					return node;
-				}
-				const visitRoutesVariable = (node: ts.Node): ts.Node => {
-					if (node.kind === ts.SyntaxKind.ArrayLiteralExpression) {
-						const array = (node as ts.ArrayLiteralExpression);
-
-						const routePath = ts.createPropertyAssignment("path", ts.createLiteral(linkPath));
-						const routeComponent = ts.createPropertyAssignment("component", ts.createIdentifier(className));
-						const routeDataInner = ts.createPropertyAssignment("text", ts.createLiteral(linkText));
-						const routeData = ts.createPropertyAssignment("data", ts.createObjectLiteral([routeDataInner]));
-						const newObject = ts.createObjectLiteral([routePath, routeComponent, routeData]);
-						this.createdStringLiterals.push(linkPath, linkText);
-
-						const notFoundWildCard = "**";
-						const nodes = ts.visitNodes(array.elements, visitor);
-						const errorRouteNode = nodes.filter(element => element.getText().includes(notFoundWildCard))[0];
-						let resultNodes = null;
-
-						if (errorRouteNode) {
-							resultNodes = nodes
-								.slice(0, nodes.indexOf(errorRouteNode))
-								.concat(newObject)
-								.concat(errorRouteNode);
+				let conditionalVisitor: NodeVisitor;
+				if (parentRoutePath === undefined) {
+					conditionalVisitor = (node: ts.Node): ts.Node => {
+						if (node.kind === ts.SyntaxKind.ArrayLiteralExpression) {
+							const newObject = this.createRouteEntry(linkPath, className, linkText);
+							const array = (node as ts.ArrayLiteralExpression);
+							this.createdStringLiterals.push(linkPath, linkText);
+							const notFoundWildCard = "**";
+							const nodes = ts.visitNodes(array.elements, visitor);
+							const errorRouteNode = nodes.filter(element => element.getText().includes(notFoundWildCard))[0];
+							let resultNodes = null;
+							if (errorRouteNode) {
+								resultNodes = nodes
+									.slice(0, nodes.indexOf(errorRouteNode))
+									.concat(newObject)
+									.concat(errorRouteNode);
+							} else {
+								resultNodes = nodes
+									.concat(newObject);
+							}
+
+							const elements = ts.createNodeArray([
+								...resultNodes
+							]);
+
+							return ts.updateArrayLiteral(array, elements);
 						} else {
-							resultNodes = nodes
-								.concat(newObject);
+							return ts.visitEachChild(node, conditionalVisitor, context);
 						}
-
-						const elements = ts.createNodeArray([
-							...resultNodes
-						]);
-
-						return ts.updateArrayLiteral(array, elements);
-					} else {
-						return ts.visitEachChild(node, visitRoutesVariable, context);
-					}
+					};
+				} else {
+					conditionalVisitor = (node: ts.Node): ts.Node => {
+						if (node.kind === ts.SyntaxKind.ObjectLiteralExpression) {
+							if (!node.getText().includes(parentRoutePath)) {
```

Comment:
That is really not a very robust way to check if that's the parent route you are looking for - the text of the whole object can contain false matches. You should be able to access `properties` on the node and look for the one with name 'path' I guess and get it's value (hint: will require interface cast to get to property assignment I think)